recipes/meson/all/conanfile.py

```diff
@@ -15,45 +16,19 @@ class MesonConan(ConanFile):
     url = "https://github.com/conan-io/conan-center-index"
     homepage = "https://github.com/mesonbuild/meson"
     license = "Apache-2.0"
-    settings = "os", "arch", "compiler", "build_type"
     no_copy_source = True
 
-    @property
-    def _settings_build(self):
-        return getattr(self, "settings_build", self.settings)
-
     def layout(self):
         basic_layout(self, src_folder="src")
 
     def requirements(self):
-        if self.conf.get("tools.meson.mesontoolchain:backend", default=False, check_type=str) in (False, "ninja"):
+        if self.conf.get("tools.meson.mesontoolchain:backend", default="ninja", check_type=str) == "ninja":
             self.requires("ninja/1.11.1")
 
-    def package_id(self):
-        self.info.clear()
-
     def source(self):
         get(self, **self.conan_data["sources"][self.version],
             destination=self.source_folder, strip_root=True)
 
-        # FIXME: https://github.com/conan-io/conan/issues/10726
-        def _fix_symlinks(root, files):
-            if not self._settings_build.os == "Windows":
-                return
-            for filename in files:
-                filename = os.path.join(root, filename)
-                if os.path.islink(filename):
-                    target = os.readlink(filename)
-                    if "/" in target:
```

Comment:
This ``if`` was never true, at least in Windows, so this ``fixing broken link`` message was never running. I tried with latest and oldest Meson versions in ``conandata.yml``

Reply:
I am sure it was, maybe it was for one of the previous meson releases, but we faced the bug for which we should apply a workaround
anyway, please avoid mixing unrelated changes in PR, migration to v2 is one thing, and removal of the workaround is another, if it's really needed. so let's exclude this change out of the PR for now.

Reply:
UPD: I think the fact you never enter the condition in your local machine is that you don't have `Developer Mode` activated, as explained in https://github.com/conan-io/conan/issues/10726. thus, symlinks are never created for you in the first place. however, for other users, they are created and they cause package to fail to be installed.

Reply:
@SSE4, my understanding is that this workaround removal is the actual fix for conan v2 since accessing self.settings/settings_build is not allowed in conan v2.
Other changes in this PR are unrelated to conan v2 migration.

Reply:
roger that. still, that logic needs to be preserved, as it fixed an actual issue people complained about. I don't have an access to Windows machine right away, but I am sure I was able to reproduce it locally in the past.

Reply:
I'd like to follow up on this. @jcar87 and I, we have been trying our best to reproduce the error, we have tried:
- Different Windows machines with and without developer mode
- Multiple Python versions, even down to 3.6 (not supported by Meson anymore, but just in case, for previous versions), to 3.10
- Multiple Meson versions, down to the lowest declared in conandata.yml to the highest declared in one, and many intermediate ones
- Trying older versions of the recipe, like reverting to the ``self.copy()`` instead of ``copy(self...)``

It seems that the actual logic for ``readlink()`` nevers return True, because Python itself while unzipping will create regular files and folders. The only possibility that came to our mind is a Windows and a Linux system sharing the same cache, which is not a supported use case anyway. And in the worst case, the simplest approach to solve this, would be to remove the ``test_cases`` folder (the one containing the symlinks) just after unzipping it.
So I'd suggest to continue with the removal (in any case, the ``source()`` method cannot apply conditional logic based on inputs like settings) unless this can be reproduced.